XMonad/Prompt.hs

```diff
@@ -228,6 +228,15 @@ class XPrompt t where
 
 data XPPosition = Top
                 | Bottom
+                -- | Prompt will be placed in the center horizontally and
+                --   in the certain place of screen vertically. If it's in the upper
+                --   part of the screen, completion window will be placed below(like
+                --   in 'Top') and otherwise above(like in 'Bottom')
+                | CenteredAt { xpCenterY :: Rational
+                             -- ^ Relative Y coordinate of center of the prompt.
```

Comment:
"relative" means relative to screen width for prompt width and relative to screen hight for y coordinate. i did't think that this could be understood in other way